dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/api/dto/MetadataBlockWithFieldsDTO.java

```diff
@@ -0,0 +1,229 @@
+package edu.harvard.iq.dataverse.api.dto;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import edu.harvard.iq.dataverse.persistence.dataset.ControlledVocabularyValue;
+import edu.harvard.iq.dataverse.persistence.dataset.DatasetField;
+import edu.harvard.iq.dataverse.persistence.dataset.DatasetFieldType;
+import edu.harvard.iq.dataverse.persistence.dataset.DatasetFieldUtil;
+import edu.harvard.iq.dataverse.persistence.dataset.DatasetFieldsByType;
+import edu.harvard.iq.dataverse.persistence.dataset.FieldType;
+import edu.harvard.iq.dataverse.persistence.dataset.MetadataBlock;
+import io.vavr.control.Option;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+
+@JsonInclude(JsonInclude.Include.NON_EMPTY)
+public class MetadataBlockWithFieldsDTO {
+    private String displayName;
+
+    private List<DatasetFieldDTO> fields;
+
+    // -------------------- GETTERS --------------------
+
+    public String getDisplayName() {
+        return displayName;
+    }
+
+    public List<DatasetFieldDTO> getFields() {
+        return fields;
+    }
+
+    // -------------------- LOGIC --------------------
+
+    public void clearEmailFields() {
+        getFields().removeIf(DatasetFieldDTO::isEmailType);
+        Object TO_REMOVE = new Object();
+        for (DatasetFieldDTO field : getFields()) {
+            if (!"compound".equals(field.getTypeClass()) || field.getValue() == null) {
+                continue;
+            }
+            Object value = field.getValue();
+            if (Map.class.isAssignableFrom(value.getClass())) {
+                Map<String, DatasetFieldDTO> restOfChildrenFields = clearEmailFields(value);
+                if (restOfChildrenFields.isEmpty()) {
+                    field.setValue(TO_REMOVE);
+                }
+                continue;
+            }
+            List<?> valueList = (List<?>) value;
+            for (Object element : valueList) {
+                if (!Map.class.isAssignableFrom(element.getClass())) {
+                    break;
+                } else {
+                    clearEmailFields(element);
+                }
+            }
+            if (valueList.isEmpty()) {
+                field.setValue(TO_REMOVE);
+            }
+        }
+        getFields().removeIf(f -> TO_REMOVE.equals(f.getValue()));
```

Comment:
IMO it should be a responsibility of DatasetFieldDTO to remove email fields in its children